crates/corro-agent/src/agent/util.rs

```diff
@@ -648,26 +658,47 @@ pub async fn process_fully_buffered_changes(
                 version: Some(version),
             })?;
 
+        let mut last_cleared: Option<Timestamp> = None;
         for (actor_id, versions_set) in overwritten {
+            if actor_id != agent.actor_id() {
+                warn!("clearing empties for another actor: {actor_id}")
+            }
             for versions in versions_set {
-                store_empty_changeset(&tx, actor_id, versions)?;
+                let ts = Timestamp::from(agent.clock().new_timestamp());
+                let inserted = store_empty_changeset(&tx, actor_id, versions, ts)?;
+                if inserted > 0 {
+                    last_cleared = Some(ts);
+                }
             }
         }
 
+        let mut agent_snap = agent_booked.snapshot();
+        if let Some(ts) = last_cleared {
+            agent_snap
+                .update_cleared_ts(&tx, ts)
```

Comment:
I don't think we want to store the last cleared ts **except** during the sync process with the originating node.

Reply:
Here, we are updating the node's own last cleared ts. Some lines before, we check for overwritten version in the `__corro_versions_impacted` table and store empty changes. This table only contains cleared versions for that node and I have added a warning if for some reason it does not.